### PR TITLE
[DPE-5114][DPE-5118] Colocated controller-broker

### PR DESCRIPTION
**Pull request by zmraul**
Initial milestone to have a co-located Kafka in KRaft mode. General fixes will be addressed on upcoming PRs. Description for the whole feature can be viewed on #232

### CODE REVIEW

File: src/core/cluster.py [hunk 1]
```diff
@@ -465,15 +466,17 @@ def _broker_status(self) -> Status:
         if not self.runs_broker:
             return Status.ACTIVE
 
-        if not self.zookeeper:
-            return Status.ZK_NOT_RELATED
+        # FIXME
```
Comment (Batalex) [inline src/core/cluster.py:469]: **suggestion**: maybe we could rename `_broker_status` to `_broker_with_zk_status` or something like that, and add a KRaft equivalent. We would then choose which one to use in `ready_to_start`

File: src/charm.py [hunk 2]
```diff
@@ -89,6 +89,11 @@ def _on_install(self, _) -> None:
             self._set_status(Status.SNAP_NOT_INSTALLED)
             return
 
+        # TODO: run on install
```
Comment (Batalex) [inline src/charm.py:92]: **question**: does it need to run post upgrade as well?

**Review by Batalex — COMMENTED** [review]
It definitely is tricky to accommodate both ZK and Kraft in the same codebase

File: src/charm.py (hunk 2 above)
Comment (zmraul) [reply src/charm.py:92]: Turns out doesn't need to run on install. This part is now moved and will no-op if dirs are already formatted

File: src/managers/config.py [hunk 3]
```diff
@@ -547,6 +560,41 @@ def metrics_reporter_properties(self) -> list[str]:
...
+    def controller_properties(self) -> list[str]:
+        """Builds all properties necessary for starting Kafka controller service.
+
+        Returns:
+            List of properties to be set
+        """
+        # TODO: this check will need to be done on relation data
+        roles = "process.roles="
+        if self.state.runs_controller:
+            roles += self.config.roles
```
Comment (Batalex) [inline src/managers/config.py:580]: **todo**: we need to remove `balancer` if it's present

```suggestion
        if self.state.runs_controller:
            roles += ",".join(set(self.config.roles.split(",")) - {"balancer"})
```

Comment (zmraul) [reply src/managers/config.py:580]: Will address in an upcoming PR

**Pull request merged**